contrib/tools/workflowcheck/README.md

```diff
@@ -0,0 +1,250 @@
+# Temporal Workflow Check
+
+Temporal Workflow Check is a tool that statically analyzes
+[Temporal Go workflows](https://docs.temporal.io/docs/go/workflows/) (i.e. functions passed to
+`worker.RegisterWorkflow`) to check for non-deterministic code either directly in the function or in a function called
+by the workflow. It is highly optimized to scan large codebases quickly.
+
+**NOTE: This will not catch all cases of non-determinism such as global var mutation. This is just a helper and
+developers should still scrutinize workflow code for other non-determinisms.**
+
+## Installing
+
+To install with [Go](https://golang.org/) installed and on the `PATH`, run:
+
+    go install go.temporal.io/sdk/contrib/tools/workflowcheck
+
+Or you can simply build by running `go build` inside of this directory. 
+
+## CLI Usage
+
+The executable has arguments in the form:
+
+    workflowcheck [-flag] [package]
+
+To simply check all source, navigate to the root of the repository/module to check and, with `workflowcheck` on
+the `PATH`, run:
+
+    workflowcheck ./...
+
+For each workflow passed to `RegisterWorkflow` that is considered non-deterministic, this will output the reasons why.
```

Comment:
```suggestion
For each Workflow Definition registered with `RegisterWorkflow` that is considered non-deterministic, this will output the reasons why.
```

Reply:
Fixed